### PR TITLE
fix: [#17] Fix launching ThinkingRock with Java17

### LINKED ISSUE #17: Compatibility with Java 11 and Java 17
As a user of thinking rock, I'd like to run the application either with Java 11 or Java 17.

### Current behavior

ThinkingRock in branch `main` runs with java11. Trying to launch it with Java 17 fails.

<details><summary>Stack-Trace</summary>
<pre>
FATAL: No way to find original stream handler for jar protocol
java.lang.reflect.InaccessibleObjectException: Unable to make field transient java.net.URLStreamHandler java.net.URL.handler accessible: module java.base does not "opens java.net" to unnamed module @5f2050f6
        at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:354)
        at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
        at java.base/java.lang.reflect.Field.checkCanSetAccessible(Field.java:178)
        at java.base/java.lang.reflect.Field.setAccessible(Field.java:172)
        at org.netbeans.ProxyURLStreamHandlerFactory.register(ProxyURLStreamHandlerFactory.java:59)
        at org.netbeans.JarClassLoader.<clinit>(JarClassLoader.java:117)
        at org.netbeans.MainImpl.execute(MainImpl.java:153)
        at org.netbeans.MainImpl.main(MainImpl.java:60)
        at org.netbeans.Main.main(Main.java:58)

java.lang.Error: factory already defined
        at java.base/java.net.URL.setURLStreamHandlerFactory(URL.java:1228)
        at org.eclipse.osgi.framework.internal.core.Framework.installURLStreamHandlerFactory(Framework.java:1765)
        at org.eclipse.osgi.framework.internal.core.Framework.<init>(Framework.java:232)
        at org.eclipse.osgi.framework.internal.core.EquinoxLauncher.internalInit(EquinoxLauncher.java:67)
        at org.eclipse.osgi.framework.internal.core.EquinoxLauncher.init(EquinoxLauncher.java:37)
        at org.eclipse.osgi.launch.Equinox.init(Equinox.java:178)
        at org.netbeans.modules.netbinox.Netbinox.init(Netbinox.java:63)
        at org.netbeans.core.netigso.Netigso.prepare(Netigso.java:146)
        at org.netbeans.NetigsoHandle.turnOn(NetigsoHandle.java:115)
        at org.netbeans.ModuleManager.enable(ModuleManager.java:1445)
        at org.netbeans.ModuleManager.enable(ModuleManager.java:1254)
        at org.netbeans.core.startup.ModuleList.installNew(ModuleList.java:316)
        at org.netbeans.core.startup.ModuleList.trigger(ModuleList.java:252)
        at org.netbeans.core.startup.ModuleSystem.restore(ModuleSystem.java:298)
        at org.netbeans.core.startup.Main.getModuleSystem(Main.java:173)
        at org.netbeans.core.startup.Main.getModuleSystem(Main.java:142)
        at org.netbeans.core.startup.Main.start(Main.java:299)
        at org.netbeans.core.startup.TopThreadGroup.run(TopThreadGroup.java:98)
        at java.base/java.lang.Thread.run(Thread.java:833)
</pre>
</details>

### Desired behavior

We should be able to start thinking rock with at least Jdk 11 or 17 installed without errors.

### Steps

The current Netbeans/Thinking-Rock setup does not work well with JPMS/Project Jigsaw. The easiest way to work around it for now is to explicitly open up reflective access to encapsulated Java APIs using the `--ad-opens` option for APIs we need reflective access to.

Beyond this issue, we should try to replace such reflective calls to encapsulated APIs, thus more and more getting rid of the need to rely on the `--add-opens` option.

- set relevant `--add-opens` option
- set java source version to 17
- set java target version to 11

### PR DESCRIPTION
**Pull request by ursjoss**
This PR makes it possible to not only run ThinkingRock with Java 11 but also with Java 17.

fixes #17

### CODE REVIEW

**Comment by johandersson** [comment]
I have tested it with Java 17 and it works. I also got it to run with Java 18 actually but I haven't tried all the functionality of course. Is this ticket just to get it to run on Java 17?

**Comment by ursjoss** [comment]
@johandersson Thanks for testing this. I would prefer keeping the ticket about java 17 for now. We can address java 18/19 in different tickets after getting the current PRs merged. 

Before we merge this one, the previous PRs #9, #13, #15, and #16 should be tested and merged, so I'm holding this one back for now.

**Review by RadoslavCap — APPROVED** [review]
Good to go.

Repeated the failure on the main branch (InaccessibleObjectException) and run properly on java 17 (17.0.5; OpenJDK 64-Bit Server VM 17.0.5+8-LTS; Red Hat, Inc.) with this code.

Nice work, I'll use 17 now, thank you.

**Pull request merged**
